infra_tooling_account/infra_tooling_account/infra_tooling_monitoring_stack.py

```diff
@@ -0,0 +1,211 @@
+        )
+
+        metrics_collection_interval_min = self.get_metrics_collection_interval_min()
+
+        rule = events.Rule(
+            self,
+            "MetricsExtractionScheduleRule",
+            schedule=events.Schedule.rate(
+                Duration.minutes(metrics_collection_interval_min)
+            ),
+            rule_name=f"rule-{self.project_name}-metrics-extract-cron-{self.stage_name}",
+        )
+        rule.add_target(targets.LambdaFunction(extract_metrics_orch_lambda))
+
+    def get_common_stack_references(self):
+        """
+        Retrieves common stack references required for the stack's operations. These include the S3 bucket for settings,
+        the ARN of the internal error SNS topic, and the Timestream database ARN.
+
+        Returns:
+            tuple: A tuple containing references to the settings S3 bucket, internal error topic, and Timestream database ARN.
+        """
+        input_settings_bucket_arn = Fn.import_value(
+            f"output-{self.project_name}-settings-bucket-arn-{self.stage_name}"
+        )
+
+        input_timestream_database_arn = Fn.import_value(
+            f"output-{self.project_name}-metrics-events-storage-arn-{self.stage_name}"
+        )
+
+        input_internal_error_topic_arn = Fn.import_value(
+            f"output-{self.project_name}-internal-error-topic-arn-{self.stage_name}"
+        )
+
+        # Settings S3 Bucket Import
+        settings_bucket = s3.Bucket.from_bucket_arn(
+            self,
+            "salmonSettingsBucket",
+            bucket_arn=input_settings_bucket_arn,
+        )
+
+        # Internal Error Topic Import
+        internal_error_topic = sns.Topic.from_topic_arn(
+            self,
+            "salmonInternalErrorTopic",
+            topic_arn=input_internal_error_topic_arn,
+        )
+
+        return settings_bucket, internal_error_topic, input_timestream_database_arn
+
+    def get_metrics_collection_interval_min(self):
+        """
+        Reads the metrics collection interval from a configuration file. The interval is defined in minutes.
+
+        Returns:
+            int: The interval in minutes for collecting metrics.
+
+        Raises:
+            Exception: If the 'metrics_collection_interval_min' key is not found in the configuration file.
+        """
+        # TODO: reuse existing settings reader
+        general_settings_file_path = "../config/settings/general.json"
+        with open(general_settings_file_path) as f:
+            general_config = json.load(f)
+            tooling_section = general_config["tooling_environment"]
+            key = "metrics_collection_interval_min"
+            if key in tooling_section:
+                return tooling_section[key]
+            else:
+                raise Exception(
+                    "metrics_collection_interval_min key not found in general.json config file ('tooling_environment' section)"
+                )
+
```

Comment:
Usually user-defined exception is used instead, but most probably all this settings part will be rewritten soon :)

Reply:
yes, it's just temporary